mne/realtime/tests/test_mockclient.py

```diff
@@ -29,16 +30,46 @@ def test_mockclient():
     data = epochs.get_data()
 
     rt_client = MockRtClient(raw)
+    # choose "large" value, should always be longer than execution time of
+    # get_data()
+    isi_max = 0.5
     rt_epochs = RtEpochs(rt_client, event_id, tmin, tmax, picks=picks,
-                         isi_max=0.5)
+                         isi_max=isi_max)
 
     rt_epochs.start()
     rt_client.send_data(rt_epochs, picks, tmin=0, tmax=10, buffer_size=1000)
 
+    # get_data() should return immediately and not wait for the timeout
+    start_time = time.time()
     rt_data = rt_epochs.get_data()
-
-    assert_true(rt_data.shape == data.shape)
+    retrieval_time = time.time() - start_time
+    assert retrieval_time < isi_max
+    assert rt_data.shape == data.shape
     assert_array_equal(rt_data, data)
+    assert len(rt_epochs) == len(epochs)
+
+    # iteration over epochs should block until timeout
+    rt_iter_data = list()
+    start_time = time.time()
+    for cur_epoch in rt_epochs:
+        rt_iter_data.append(cur_epoch)
+    retrieval_time = time.time() - start_time
+    assert retrieval_time >= isi_max
+    rt_iter_data = np.array(rt_iter_data)
+    assert rt_iter_data.shape == data.shape
+    assert_array_equal(rt_iter_data, data)
+    assert len(rt_epochs) == len(epochs)
+
+    export_file = tmpdir.join('test_rt-epo.fif')
+    try:
+        rt_epochs.save(str(export_file))
+
+        loaded_epochs = mne.read_epochs(str(export_file))
+        loaded_data = loaded_epochs.get_data()
+        assert rt_data.shape == loaded_data.shape
+        assert_allclose(loaded_data, rt_data)
+    finally:
+        export_file.remove(ignore_errors=True)
```

Comment:
Can we avoid doing this? Based on what you say the `tmpdir` will eventually go away on its own. We don't want to have to add these all over our code, `_TempDir` does not require it.